### ISSUE TITLE
Unable to get log

### ISSUE THREAD

**Issue by milos-matijasevic** [kubernetes-sigs/controller-runtime#1033]
When I import "sigs.k8s.io/controller-runtime/pkg/log" I get this error 
```
../sigs.k8s.io/controller-runtime/pkg/log/null.go:48:32: undefined: logr.InfoLogger
```
I assume here was the same problem https://github.com/kubernetes/klog/issues/163, due to logr refactoring and removing InfoLogger https://github.com/go-logr/logr/pull/16

**Comment by RafalSkolasinski** [comment]
We were just hit by the same when trying to update our dependencies.

**Comment by milos-matijasevic** [comment]
Should i make pull request changing InfoLogger to Logger, or you will deal with it?

**Comment by RafalSkolasinski** [comment]
@milos-matijasevic I am not maintainer of this package but I guess they will be happy if you submit a PR :-)

**Comment by milos-matijasevic** [comment]
@RafalSkolasinski Actually there is already pull request for fixing this https://github.com/kubernetes-sigs/controller-runtime/pull/1011

**Comment by adriangonz** [comment]
An alternative could be to add back `InfoLogger` into `go-logr/logr@v0.2.1`, which this PR is already trying to do: https://github.com/go-logr/logr/pull/20

Once that PR is merged (if it gets merged) we could safely update the dependency on `go-logr/logr` to `0.2.1`. Since this wouldn't be a breaking change anymore, could that be included in the next non-breaking release? When is the next non-breaking release planned for?

BTW it seems that #1011 is a duplicate of #995.

**Comment by vincepri** [comment]
This will be fixed in Kubernetes 1.19 release (v0.7.x) which is due ~1-2 months

**Comment by jpeach** [comment]
AFAICT this is still a problem with Kubernetes 1.19 and controller-runtime 0.6.x

```
$ cat go.mod
module github.com/projectcontour/contour

go 1.15

require (
	github.com/ahmetb/gen-crd-api-reference-docs v0.1.5
	github.com/alecthomas/template v0.0.0-20190718012654-fb15b899a751 // indirect
	github.com/alecthomas/units v0.0.0-20190717042225-c3de453c63f4 // indirect
	github.com/client9/misspell v0.3.4
	github.com/envoyproxy/go-control-plane v0.9.6
	github.com/golang/protobuf v1.4.2
	github.com/google/go-cmp v0.5.0
	github.com/google/uuid v1.1.1
	github.com/grpc-ecosystem/go-grpc-prometheus v1.2.0
	github.com/konsorten/go-windows-terminal-sequences v1.0.2 // indirect
	github.com/prometheus/client_golang v1.1.0
	github.com/prometheus/client_model v0.2.0
	github.com/prometheus/common v0.6.0
	github.com/sirupsen/logrus v1.4.2
	github.com/stretchr/testify v1.5.1
	google.golang.org/grpc v1.27.0
	google.golang.org/protobuf v1.25.0
	gopkg.in/alecthomas/kingpin.v2 v2.2.6
	gopkg.in/yaml.v2 v2.3.0
	k8s.io/api v0.19.2
	k8s.io/apimachinery v0.19.2
	k8s.io/client-go v0.19.2
	k8s.io/klog v1.0.0
	k8s.io/utils v0.0.0-20200729134348-d5654de09c73
	sigs.k8s.io/controller-runtime v0.6.3
	sigs.k8s.io/controller-tools v0.2.9
	sigs.k8s.io/kustomize/kyaml v0.1.1
	sigs.k8s.io/service-apis v0.0.0-20200213014236-51691dd89266
)

$ make check
go install -mod=readonly -v -ldflags="-s -w -X github.com/projectcontour/contour/internal/build.Version=1934ce5c -X github.com/projectcontour/contour/internal/build.Sha=1934ce5c95d9ed7484858dd1d91cf5fe858b95b7 -X github.com/projectcontour/contour/internal/build.Branch=use-consistent-informers" -tags "oidc gcp" github.com/projectcontour/contour/cmd/contour
sigs.k8s.io/controller-runtime/pkg/log
# sigs.k8s.io/controller-runtime/pkg/log
../../go/pkg/mod/sigs.k8s.io/controller-runtime@v0.6.3/pkg/log/null.go:48:32: undefined: logr.InfoLogger
make: *** [Makefile:79: install] Error 2
```

The fix (at least for me) seems to be:
```
$ go get -u github.com/go-logr/logr
...
$ git diff go.mod
diff --git go.mod go.mod
index ec8d9661..3d04cea5 100644
--- go.mod
+++ go.mod
@@ -8,6 +8,7 @@ require (
        github.com/alecthomas/units v0.0.0-20190717042225-c3de453c63f4 // indirect
        github.com/client9/misspell v0.3.4
        github.com/envoyproxy/go-control-plane v0.9.6
+       github.com/go-logr/logr v0.2.1 // indirect
        github.com/golang/protobuf v1.4.2
        github.com/google/go-cmp v0.5.0
        github.com/google/uuid v1.1.1
```

**Comment by vincepri** [comment]
@jpeach We have some alpha tags for v0.7 to test out some of these changes, https://github.com/kubernetes-sigs/controller-runtime/releases — ⚠️ though, there are a lot of other changes that need to be documented before release, proceed with caution :) 

The update to 1.19 and the new go-logr dependency is going to come in when v0.7 gets released

**Comment by alvaroaleman** [comment]
@jpeach also, controller-runtime doesn't support using an arbitrary version of the kube dependencies. It might work, but its untested and not officially supported. The 0.6.X tree supports the 1.18 version of the kube dependencies: https://github.com/kubernetes-sigs/controller-runtime/blob/ac380d61764a160b32946e606b0c9ecd2834e3e8/go.mod#L27

Hence, this is isn't really a bug in controller-runtime.

**Comment by estroz** [comment]
This was fixed by #1170 so can this be closed @vincepri

**Comment by camilamacedo86** [comment]
Hi @milos-matijasevic, 

I am closing this one based on the comment above https://github.com/kubernetes-sigs/controller-runtime/issues/1033#issuecomment-721302464.  However, feel free to ping us to re-open if you think that it is required.
